partiql-tests-data/success/syntax/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,402 @@
+// SQL style
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - No Second Precision ",
+  statement: "TIMESTAMP '2023-06-01 00:00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Arbitrary Second Precision",
+  statement: "TIMESTAMP '2023-06-01 00:00:00.00000000000",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision = Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision < Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00.0000",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision > Digit in Second Fraction",
+  statement: "TIMESTAMP(10) '2023-06-01 00:00:00.0000",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITH TIME ZONE - SQL Style - No Second Precision ",
+  statement: "TIMESTAMP '2023-06-01 00:00:00+00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITH TIME ZONE - SQL Style - Arbitrary Second Precision",
+  statement: "TIMESTAMP '2023-06-01 00:00:00.00000000000+00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITH TIME ZONE - SQL Style - Precision = Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00+00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITH TIME ZONE - SQL Style - Precision < Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00.0000+00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITH TIME ZONE - SQL Style - Precision > Digit in Second Fraction",
```

Comment:
similar to above comment regarding special characters. also applies to below tests using RFC3339.